src/backends/plonky2/circuits/common.rs

```diff
@@ -63,6 +64,15 @@ impl ValueTarget {
     }
 }
 
+/// StatementArgTarget to ValueTarget coercion. Make sure to check
+/// that the arg is a value using the `statement_arg_is_value` method
+/// first!
+impl From<&StatementArgTarget> for ValueTarget {
+    fn from(st_arg: &StatementArgTarget) -> Self {
```

Comment:
This seems a bit dangerous.
What about adding a method to `StatementArgTarget` that does this instead of using the `From` trait?
I fear misusing this if we keep it like this.